### PR TITLE
Add Rails support to Notebook, via notebook_attachment macro

### PR DESCRIPTION
**Pull request by maclover7**
cc @tute

### CODE REVIEW

File: .rubocop_todo.yml [hunk 1]
```diff
@@ -4,3 +4,7 @@ Style/DoubleNegation:
   Enabled: false
 Style/LineLength:
   Max: 110
+Style/MethodLength:
+  Max: 15
```
Comment (tute) [inline .rubocop_todo.yml]: Woot! :)

This is the thing with failing the build through rubocop, we need to relax the rules. A much better rule is `5`, with some agreed upon exceptions, that are exceptional. I wouldn't commit this, and either stop failing the build, or ignore the build for this PR when it's only that rule.

File: lib/notebook/rails/active_record.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
+require 'active_record'
+
+module Notebook
+  module ActiveRecordAttachment
+    extend ActiveSupport::Concern
```
Comment (tute) [inline lib/notebook/rails/active_record.rb]: Style: I would leave a new line between this and the next one.

File: lib/notebook/rails/active_record.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    module ClassMethods
+      # Attachment method which hooks into ActiveRecord models
+      def notebook_attachment(name)
+        define_method "#{name}" do
+          return @notebook_attachment
```
Comment (tute) [inline lib/notebook/rails/active_record.rb]: We can use ruby's implicit `return` statements, and remove it.

File: spec/rails/active_record_spec.rb [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  let(:klass) do
+    # opts = options
+    Class.new(ActiveRecord::Base) do
+      self.table_name = :posts
+      def self.name
+        'Post'
+      end
+      notebook_attachment :document
+    end
+  end
```
Comment (tute) [inline spec/rails/active_record_spec.rb]: Very nice specs, which don't need a "dummy" app. :+1:

File: spec/spec_helper.rb [hunk 5]
```diff
@@ -23,3 +23,34 @@ def self.fixture_directory
     FileUtils.mkdir 'spec/fixtures/public'
   end
 end
+
+module Refile
```
Comment (tute) [inline spec/spec_helper.rb]: It looks like this is not used? Should we remove it?

**Comment by tute** [comment]
Looking good!

File: spec/spec_helper.rb (hunk 5 above)
Comment (maclover7) [inline spec/spec_helper.rb]: my bad, copied this implementation from another gem, but isn't needed.

**Comment by maclover7** [comment]
Pushed and updated. Will merge on green Travis CI build.

**Comment by maclover7** [comment]
`before_upload` and `after_upload` Rails helpers will be up in next PR, stay tuned!

**Pull request merged**